syntax = "proto3";
package instance;




import "google/protobuf/struct.proto";

/*
PostSearch请求
*/
message PostSearchRequest {
    /*
    模型对象ID
    */
    string objectId = 1 ;
    /*
    e.g.: { name: { $like: '%q%' } }, { $or: [{ name: { $like: '%q%' }}] }
    */
    google.protobuf.Struct query = 2 ;
    /*
    e.g.: { instanceId: true, name: true }
    */
    google.protobuf.Struct fields = 3 ;
    /*
    当为 true 时，只搜索与我相关实例
    */
    bool only_my_instance = 4 ;
    /*
    对于关联的实例数据是否只获取 relation_view 中指定的属性, 这个字段为 true 时, 会覆盖 fields 字段中指定的二级字段设置
    */
    bool only_relation_view = 5 ;
    /*
    页码
    */
    int32 page = 6 ;
    /*
    页大小
    */
    int32 page_size = 7 ;
    /*
    按字段排序, 留空默认按照实例ID降序排序(1表示升序, -1表示降序) e.g.: { instanceId: 1 }
    */
    google.protobuf.Struct sort = 8 ;
    /*
    按照权限过滤(通用实例都有 read, update, delete 权限控制, 主机实例在通用实例权限基础上有额外的 operate 权限, 应用实例在通用实例权限基础上有额外的 developClusterOperate, testClusterOperate, prereleaseClusterOperate, productionClusterOperate 权限) e.g.: [ "operate", "update" ]
    */
    repeated string permission = 9 ;
}

/*
PostSearch返回
*/
message PostSearchResponse {
    /*
    instance list
    */
    repeated google.protobuf.Struct list = 1 ;
    /*
    实例总数
    */
    int32 total = 2 ;
    /*
    页码
    */
    int32 page = 3 ;
    /*
    页大小
    */
    int32 page_size = 4 ;
}

/*
PostSearchApi返回
*/
message PostSearchResponseWrapper {
    /*
    返回码
    */
    int32 code = 1 ;
    /*
    返回码解释
    */
    string codeExplain = 2 ;
    /*
    错误详情
    */
    string error = 3 ;
    /*
    返回数据
    */
    instance.PostSearchResponse data = 4 ;
}

